projects/web-wallet/wallet-connect/0001-design.md

```diff
@@ -0,0 +1,774 @@
+      requiredMethods: params.requiredNamespaces.hathor.methods,
+    },
+    onAcceptAction,
+    onRejectAction,
+  }));
+
+  const { accept } = yield race({
+    accept: take(onAcceptAction.type),
+    reject: take(onRejectAction.type),
+  });
+
+  if (!accept) {
+    yield call(() => web3wallet.rejectSession({
+      id: params.id,
+      reason: {
+        code: REJECTION_CODE,
+        message: 'User rejected the session',
+      },
+    }));
+  }
+
+  try {
+    const sessionApproved = yield call(() => web3wallet.approveSession({
+      id,
+      relayProtocol: params.relays[0].protocol,
+      namespaces: {
+        hathor: {
+          accounts: accept.payload.selectedAddresses.map((address) => `hathor:mainnet:${address}`),
+          chains: [`hathor:${network}`],
+          events: [],
+          methods: HATHOR_ALLOWED_METHODS,
+        },
+      },
+    }));
+
+    yield call(refreshActiveSessions);
+  } catch(e) {
+    console.log('Error handling session proposal: ', e);
+  }
+}
+```
+
+In this example, we are displaying a modal describing the methods that the `dApp` is requesting and after approal, it sends an `session_approved` method through the `web3wallet` instance.
+
+The modal must display a list of the wallet's addresses and allow the user to select one or more of them to connect (see the design decisions section)
+
+<a name="reference_level_explanation__handling_session_requests"/>
+
+### Handling session requests
+
+After the session is established, we will receive RPC requests through the `session_request` event, here is an example request:
+
+```json
+{
+    "id": 1679071024881602,
+    "topic": "d5b50afb3541aea8ce9c8a0341a18bff913d3d83602084424cffcd3b81c28626",
+    "params": {
+        "request": {
+            "method": "hathor_signMessage",
+            "params": {
+                "address": "HRZZrkp3BcCdvL5RFd4sb3BqXienb5V5yr",
+                "message": "Please sign me!"
+            }
+        },
+        "chainId": "hathor:mainnet"
+    }
+}
+```
+
+As we can see, in the params we can see the requested method and the params, in this case, the address to sign with and the message for the `signMessage` RPC API
+
+Handling this specific message will be described in the API section of the reference-level explanation
+
+### API format, parameters and response
+
+#### 1. Prove ownership of an address
+
+When requested, this method should display a modal asking user for confirmation before continuing. In this case, it should display the `message` and the `address` the `dApp` is requesting it to sign
+
+This method should validate if the requested address belongs to the current wallet (with the limitation of only validating if the address is reachable with the fullnode's current `MAX_GAP`), retrieve its `privateKey` and sign the message, returning a base64 signed result. 
+
+**Params**:
+* `address`: Base58 address to request ownership proof
+* `message`: Arbitrary message to sign
```

Comment:
I'd say "prefixed message to be signed (e.g., htr-message: XXX)".

Reply:
Added an explanation on this on this commit: e330ec6d36986cf679b7a3b462d54b684afa4c8e

Reply:
Thanks!